### ISSUE TITLE
Bug #1: VAEGAN module loss function

### ISSUE THREAD

**Issue by Ram81** [Ram81/AC-VAEGAN-PyTorch#5]
Current implementation of the AC VAEGAN model is not able to converge properly. After investigating the issue we found some bugs in implementation of loss function.

```
        # reconstruction errors, not used as part of loss just to monitor
        nle = 0.5 * (ten_original.view(len(ten_original), -1)) - ten_predict.view((len(ten_predict), -1)) ** 2

        # kl-divergence
        kl = -0.5 * torch.sum(-variances.exp() - torch.pow(mu, 2) + variances + 1, 1)

        # mse between intermediate layers
        mse = torch.sum((layer_original - layer_predicted) ** 2, 1)

        # BCE for decoder & discriminator for original, sampled & reconstructed
        # the only excluded is the bce_gen original

        bce_dis_original = -torch.log(labels_original)
        bce_dis_sampled = -torch.log(1 - labels_sampled)

        bce_gen_original = -torch.log(1 - labels_original)
        bce_gen_sampled = -torch.log(labels_sampled)

        aux_criteron = nn.NLLLoss()
        nllloss_aux_original = aux_criteron(aux_labels_predicted, aux_labels_original)
        nllloss_aux_sampled = aux_criteron(aux_labels_sampled, aux_labels_original)
````

Please refer the above code. Full source code is available in `model.py`.
